specification/metadata-3.2.md

```diff
@@ -45,151 +44,174 @@ of `METADATA LIST` MUST be as follows:
-does not possess.
+It is an error for users to set keys on targets for which they lack
+authorization from the server, and the server MUST respond with
+`ERR_KEYNOPERMISSION`.
 
-*Errors*: ERR_KEYINVALID, ERR_KEYNOTSET, ERR_KEYNOPERMISSION
+*Errors*: `ERR_METADATALIMIT`, `ERR_KEYINVALID`, `ERR_KEYNOTSET`, `ERR_KEYNOPERMISSION`
 
 ### METADATA CLEAR
 
-This subcommand MUST remove all metadata, equivalently to using METADATA SET
-on all keys with an empty value. The format of `METADATA CLEAR` MUST be as
-follows:
+This subcommand MUST remove all metadata, equivalently to using `METADATA SET`
+on all currently-set keys with an empty value. The format of `METADATA CLEAR`
+MUST be as follows:
 
 `METADATA <Target> CLEAR`
 
-This subcommand has the same conditions and errors as the `METADATA SET`
-subcommand.
+The server MUST respond with one `RPL_KEYVALUE` event per cleared key and one
+`RPL_METADATAEND` event. It is an error for users to use this subcommand on
+targets for which they lack authorization from the server. Servers MAY reject
+this subcommand for channels, using `ERR_KEYNOPERMISSION` with an asterisk
+(`*`) in the `<Key>` field.
 
 ## Metadata Notifications
 
-Notifications, e.g. from metadata subscriptions, MUST be passed to the client
-using the incoming METADATA verb, the format of which is as follows:
+Metadata notifications are enabled by requesting the `metadata-notify`
+capability during capability negotiation. When negotiated, this capability
+extends `MONITOR` behaviour to include subscribing users to notifications for
+those users they are currently monitoring. Clients are also subscribed to
+notifications for channels they join. Clients may discontinue notifications
+for users by issuing a disabling `MONITOR` command, and for channels by
+parting the channel. Clients are automatically subscribed to notifications for
+their own metadata.
+
+Notifications use the `METADATA` event, the format of which is as follows:
+
+`METADATA <Target> <Key> :<Value>`
 
-`METADATA <Source> <Target> <Key> :<Value>`
+`<Target>` refers to the entity which had its metadata changed.
 
-`<Source>` identifies the source of the change. `<Target>` refers to the target
-specified by the `MONITOR` command.
+Clients MUST handle all metadata notifications, whether they explicitly
+requested them or not.
 
-This specification extends `MONITOR` to enable metadata notifications using the
-above verb. Clients MUST handle all `METADATA` verbs, regardless of
-subscription status. Notifications MUST be sent for the client's own keys,
-regardless of subscription status.
+Metadata propagates to clients automatically under certain conditions:
+
+1. Upon authentication and successful negotiation of `metadata-notify`, clients
+   MUST have their non-transient metadata propagated to them. If none exists,
+   servers MUST send `ERR_METADATAEND`.
+2. Clients SHOULD have current metadata propagated to them.
```

Comment:
I'm puzzled as to why this comment was applied to this portion of the diff, but clients are only sent metadata notifications for changes they themselves didn't make.